sympy/integrals/integrals.py

```diff
@@ -1404,6 +1406,79 @@ def integrate(*args, **kwargs):
     else:
         return integral
 
+def cauchy_principal_value(func, sym, lower_limit = -oo, upper_limit = oo):
```

Comment:
This is a special method for evaluating integrals. It should be a method of `Integral`.

Reply:
@jksuom  Sorry I didn't get this point. Do you mean I should add this as a method somewhere else or I should have used something else?
Please clarify.

Reply:
I think this should be in `class Integral`, like `def cauchy_principal_value(self)`.

Reply:
@jksuom 
>I think this should be in `class Integral`, like `def cauchy_principal_value(self).`

I searched but couldn't find an existing function of that sort. I guess it isn't there yet. Although I could be wrong too. Let me know if you find it, please.

Reply:
That is the method to be defined (by this function).

Reply:
@jksuom Okay so I should add this to `class Integral` right ?
I'll do it.

Reply:
@jksuom  I'll be submitting another commit soon where there will be all the changes you asked for. I'm currently working on the algorithm of breaking the integral on its singularities and then summing up the resulting (child) integrals.

Reply:
@jksuom  Do you think such an algorithm is already present in some other function which I could use? Or maybe it's been implemented in `diofant` (so that I could take a look at that implementation) ?